### ISSUE TITLE
Issue with multiple registries installing from shrinkwrap

### ISSUE THREAD

**Issue by nickpape-msft** [pnpm/pnpm#1076]
### Code to reproduce the issue:

Setup an `.npmrc` file that points at two registries:

```
@foo:registry=https://dev.pkgs.visualstudio.com/_packaging/ScopedFeed/npm/registry/
//dev.pkgs.visualstudio.com/_packaging/ScopedFeed/npm/registry/:always-auth=true

registry=https://dev.pkgs.visualstudio.com/_packaging/RegularFeed/npm/registry/
//dev.pkgs.visualstudio.com/_packaging/RegularFeed/npm/registry/:always-auth=true
```

Run "pnpm install" without a shrinkwrap. Everything works.

Delete your local folder and run "pnpm install" again.

### Expected behavior:

The correct URL is resolved.

### Actual behavior:

PNPM tries to download from the wrong registry.

E.g. 

```
 ERROR  Fetching dev.pkgs.visualstudio.com/@foo/bar/1.0.0 failed!
 ERROR  404 Not Found: https://dev.pkgs.visualstudio.com/_packaging/RegularFeed/npm/registry/@foo/bar/-/bar-1.0.0.tgz
```

It looks like PNPM is only considering the `registry` listed in the shrinkwrap, but is not using the `.npmrc` to check which registry should be used with a particular scope. I tried changing the registry in the shrinkwrap and confirmed this.

### Additional information:

 - `node -v` prints: v8.9.4
 - Windows, OS X, or Linux?: Windows

**Comment by nickpape-msft** [comment]
Obviously we advise people not to use multiple registries, but it appears some users of Rush already have been. This seems to be blocking their migration off npm and to pnpm.

**Comment by zkochan** [comment]
Could you please try with the latest pnpm? I fixed something related in [pnpm@1.35.9](https://github.com/pnpm/pnpm/releases/tag/v1.35.9)

**Comment by nickpape-msft** [comment]
Still appears to be occurring even with 1.35.10.

**Comment by zkochan** [comment]
have you recreated the `shrinkwrap.yaml` with `pnpm@1.35.10`? The issue really seems like what I fixed in `pnpm@1.35.9`. The fix was to add the tarball URL to the resolution field of the scoped package but you have to recreate the shrinwrap.yaml

**Comment by thomabr** [comment]
Hey Zoltan,

That seems to have fixed the problem. Thanks Nick for starting the conversation!
